spring-cloud-sleuth-zipkin/src/main/java/org/springframework/cloud/sleuth/zipkin2/ZipkinAutoConfiguration.java

```diff
@@ -87,9 +95,32 @@
 	public Reporter<Span> reporter(ReporterMetrics reporterMetrics,
 			ZipkinProperties zipkin, @Qualifier(SENDER_BEAN_NAME) Sender sender) {
 		// historical constraint. Note: AsyncReporter supports memory bounds
-		return AsyncReporter.builder(sender).queuedMaxSpans(1000)
+		AsyncReporter<Span> asyncReporter = AsyncReporter.builder(sender)
+				.queuedMaxSpans(1000)
 				.messageTimeout(zipkin.getMessageTimeout(), TimeUnit.SECONDS)
 				.metrics(reporterMetrics).build(zipkin.getEncoder());
+		CheckResult checkResult = checkResult(asyncReporter);
+		if (log.isDebugEnabled()) {
+			log.debug("Check result of the async reporter is [" + checkResult + "]");
+		}
+		return asyncReporter;
+	}
+
+	private CheckResult checkResult(AsyncReporter<Span> asyncReporter) {
+		ExecutorService executor = Executors.newSingleThreadExecutor();
+		Callable<CheckResult> task = asyncReporter::check;
+		Future<CheckResult> future = executor.submit(task);
+		try {
+			return future.get(1, TimeUnit.SECONDS);
+		} catch (Exception ex) {
+			if (log.isDebugEnabled()) {
```

Comment:
if this is only at startup, I think people talking about this will want it to be something higher than debug, like warn